build-system/tasks/amp-task-runner.js

```diff
@@ -112,6 +112,13 @@ async function runTask(taskName, taskSourceFilePath, taskFunc) {
  * @param {string} taskSourceFileName
  */
 function createTask(taskName, taskFuncName, taskSourceFileName) {
+  // By default, assume the function is the camelCase version of the task name.
+  taskFuncName =
+    taskFuncName ||
+    taskName.replace(/-([a-z])/g, (unusedMatch, c) => c.toUpperCase());
+  // By default, assume the task is implemented in a file of the same name.
+  taskSourceFileName = taskSourceFileName || taskName;
```

Comment:
Alternative to lines 115-120:
```js
 * @param {string} taskName
 * @param {string=} taskFuncName
 * @param {string=} taskSourceFileName
 */
function createTask(
  taskName,
  taskFuncName = taskName,
  taskSourceFileName = taskName
) {
```

Reply:
As written, this would only work for non-hyphenated tasks. Case conversion is quite common throughout AMP (I took this logic directly from `src/string`, unsure if it was okay to import a `src` module in `build-system`. What about it concerns you?

Reply:
Thanks for the context and for the pointer to `src/string.js`, very useful.

> What about it concerns you?

My main objection isn't about where or how to do automatic case conversion for task names, it is about whether to do it in the first place. The definitions in `amp.js` are meant to self-document three pieces of data (task, function, file), and I think we should continue to retain the documentation when any one of those pieces of data is not the same.

Since `createTask()` is only called from within `amp.js`, and exactly once per task, I don't think there's enough of a benefit to optimizing it via auto case conversion. That said, I agree that repetition of identical info can be eliminated where possible via default args.

Hence, the proposed compromise:

```js
createTask('ava'); // All three pieces of data are the same
createTask('bundle-size', 'bundleSize'); // Explicitly documents difference between task and function name
createTask('default', 'defaultTask', 'default-task'); // Explicitly documents all three pieces of data
```

> unsure if it was okay to import a `src` module in `build-system`

Yep, good call, we'd like to keep them independent of each other (`dep-check` may prevent this in any case).